### ISSUE TITLE
Banned accounts should not be able to post in chat or posts

### ISSUE THREAD

**Issue by ednapiranha** [bigboringsystem/bigboringsystem#143]

**Comment by ednapiranha** [comment]
once they try to post and a banned record is found, force them to log out.

**Comment by jasonrhodes** [comment]
Not sure how to test this in production but it seems to me when I ban a local user, they can log in anyway. Status is still banned for them (so my chat/post blocks are still in effect) but they are free to log out and back in. Obviously that's not what is wanted but is that even expected behavior right now?

I put a fix in this ticket, it wasn't too hard (the chat blocking was the hardest part). I just need to write up the tests and then I'll submit the PR.

**Comment by ednapiranha** [comment]
When you're banning right now, you're banning by phone # not ip. When you authenticate, it is looking for ip not phone number :) So there's a mixup there right now. I guess banning on both situations is fine and then it's tough to remove a ban on ip.. b/c we would have to associate the ip with the profile. But that could change so it gets weird.

**Comment by jasonrhodes** [comment]
Aha, makes sense. I vote for banning on phone for now and if people are circumventing with throw-away phone numbers then maybe you need a super-ban and then auth checks for both, or something like that...

**Comment by ednapiranha** [comment]
Yeah banning on phone is a problem because you could send requests to someone else's number over and over to get that auto banned.
